src/components/generic/Select/Select.jsx

```diff
@@ -89,6 +89,7 @@ export default function Select({
           onSelect={(option) => handleOptionSelect(option)}
           selected={value}
           tabIndex={tabIndex}
+          onBlur={onOutsideClick}
```

Comment:
although the effect is the same, on blur should not call a function dedicated for outside clicks. The reason is that, in the future, someone might  add/change code inside `onOutsideClick` and that should not affect what happens when the option list is blurred.

I would advise renaming the function `onOutsideClick` to `collapse`.